core/standards.md

```diff
@@ -0,0 +1,93 @@
+
+Core contracts should follow established coding patterns including repo and contract structures, variable naming patterns (yes [Daiwanese](https://docs.makerdao.com/other-documentation/system-glossary) is a standard for our team) and auth patterns. For most contracts, compiler version should be fixed at 0.8.n or 0.6.12 and optimizations should be off.  While licensing is considered on a case by case basis, it should always be declared and we tend to default to AGPLv3. We value security over gas optimizations and simplicity over complexity. As such, low level calls should be used minimally and gas optimization tricks should be used sparingly.
+
+Whenever possible, utilize an existing solution instead of building a new contract. MakerDAO is a very complex project and each new contract or derivation increases the cognitive and maintenance overhead of the entire protocol. By limiting the number of moving parts we reduce the risk of unexpected behavior.
+
+This repo attempts to document our standards, other sources include (non-exhaustive list):
+
+- [MIPS best practices](https://github.com/makerdao/mips-best-practices)
+
+## Coding standards we expect to see in core contracts
+
+### Permissions
+
+Permissioning in the contract should be handled by `wards` and the `wards` mapping should occupy storage slot `0x0`
+
+### Contract File
+
+`.sol` files should adhere to the following standards:
+
+- Line break at the end
+- Include a license at the top
+- Declare pragma version after the license
+- In almost all cases there should only be one contract per file
+
+### Contract ordering
+
+Following and expanding on [Solidity Order of layout recommendations](https://docs.soliditylang.org/en/stable/style-guide.html#order-of-layout) our contracts are generally laid out in this way:
+
+1. State variables (including wards and math)
+
+    a. Constants then immutables then storage variables. All constants and variables should have explicit visibility declared. In most cases, constants and immutables can be declared internal.
+
+    b. Wards should be storage slot `0x0`
+2. Events
+3. Getter functions (i.e. to make reading structs easier)
+
+    a. these may be placed at the end of the file if they are extensive to improve readability
+4. Constructor
+5. Modifiers
+6. Internal Math constants and functions
+7. Wards functions (i.e. `rely`/`deny`)
+8. Fileable functions
+9. Other Internal functions
+10. Other External functions
+
+### Spacing
+
+We align variables with extra whitespace. You will get PR comments if you are off by even one space.
+
+### Naming
+
+We prefer to try to have **Daiwanese** naming patterns (3 or 4 letter variables and functions) such as `guy`, `usr`, or `who` for address parameters and `wad` or `amt` for `uint` or `int` parameters (with care given to `RAY`, `RAD`, and `WAD` precision).
+
+If we are avoiding variable shadowing, we add a `_` to the end of the variable names.
```

Comment:
```suggestion
Variables named to avoid shadowing are appended with an underscore (`_`).
```
Can add here "it is standard practise for the parameters to have an underscore (basically we dont want them mixing it up?)